textgrad/engine/ollama.py

```diff
@@ -0,0 +1,100 @@
+try:
+    from ollama import Client
+    # from openai import OpenAI
+except ImportError:
+    raise ImportError("If you'd like to use OpenAI models, please install the openai package by running `pip install openai`, and add 'OPENAI_API_KEY' to your environment variables.")
+
+import os
+import platformdirs
+from tenacity import (
+    retry,
+    stop_after_attempt,
+    wait_random_exponential,
+)
+import json
+from .base import EngineLM, CachedEngine
+from openai._types import NotGiven
+
+class ChatOllama(EngineLM, CachedEngine):
+    DEFAULT_SYSTEM_PROMPT = "You are a helpful, creative, and smart assistant."
+
+    def __init__(
+        self,
+        model_string="qwen2:0.5b",
+        system_prompt=DEFAULT_SYSTEM_PROMPT,
+        **kwargs):
+        """
+        :param model_string:
+        :param system_prompt:
+        """
+        root = platformdirs.user_cache_dir("textgrad")
+        cache_path = os.path.join(root, f"cache_ollama_{model_string}.db")
+
+        super().__init__(cache_path=cache_path)
+
+        self.system_prompt = system_prompt
+        # if os.getenv("OPENAI_API_KEY") is None:
+        #     raise ValueError("Please set the OPENAI_API_KEY environment variable if you'd like to use OpenAI models.")
+        
+        self.client = Client(
+            # api_key="ollama",
+            host="http://localhost:11434",
```

Comment:
should we make this a parameter / the port some environment variable and do the associated checks? We can check environment variables to see "OLLAMA_PORT" and throw errors if it does not exist or something. just trying to ensure we are flexible / communicate to the user adequately

Reply:
"http://localhost:11434" is the default across all ollama installations.

Reply:
thanks so much for your hard work! Not all users will use localhost as the address and some may not be able to bind that port. ollama seems to have a section discussing that: https://github.com/ollama/ollama/blob/main/docs/faq.md#how-can-i-expose-ollama-on-my-network

As @mertyg suggested, i'd either make these env variables or find another way of passing arguments